### DISCUSSION TITLE
Is it possible to install a python module?

### DISCUSSION THREAD

**Discussion by knajjars** [jupyterlite/jupyterlite#548] [General]
### Problem

I want to embed with an iFrame the Jupyter REPL in a static docs website and have a third party SDK pre-installed in it.

### Additional context

I tried:
* adding the package to the `requirements.txt` file
* installing it directly in the REPL with `piplite` (not ideal solution). Also it seems that it did not work as expected, because the API requests inside our SDK failed.

I am wondering if JupyterLite is the correct tool for what we are trying to achieve?

**Comment by bollwyvl** [comment]
Putting your package as a noarch .whl in the ./pypi/ relative to where you
build should make it available for piplite to install after running
"jupyter lite build". Otherwise, loading it by url would work.

You would then use the ?code params to install it with piplite in a user
visible way (important dude if/when something goes wrong).

Http requests with stdlib or requests will probably not work and would need
to be shimmed to use js.fetch or pyodide's built-in fetch.

As the owner of your package, you could:
- provide a way to inject another client
- do a custom build just for lite

**Comment by jobovy** [comment]
@knajjars: This is related to #508 and [this comment](https://github.com/jupyterlite/jupyterlite/issues/508#issuecomment-1051479045) may be helpful to you.

**Comment by bollwyvl** [comment]
As this is still rather light on details/feedback, and it seems to be about a _specific_ python module, i'm moving this to a discussion topic.
